### PR TITLE
[#2] connections with provider certificate

### LINKED ISSUE #2: Connecting to APNs with Provider Certificates
get a connection with APNs with Certificates

### PR DESCRIPTION
**Pull request by ferigis**
[close #2]

### CODE REVIEW

File: lib/pushito/connection/state.ex [hunk 1]
```diff
@@ -0,0 +1,10 @@
+defmodule Pushito.Connection.State do
+  @moduledoc """
+  This module represents the connection state for Pushito.Connection
+  """
+
+  @enforce_keys [:config, :client, :http2_connection]
```
Comment (cabol) [inline lib/pushito/connection/state.ex]: Suggestion: I would change `: http2_connection ` by `: h2_connection `

File: test/pushito_test.exs [hunk 2]
```diff
@@ -2,7 +2,30 @@ defmodule PushitoTest do
...
+    {:ok, _} = Application.ensure_all_started(:pushito)
+
+    on_exit fn ->
+      Application.stop(:pushito)
+    end
+
+    :ok
+  end
+
+  test "make a connection to APNs with Provider Certificates" do
```
Comment (cabol) [inline test/pushito_test.exs:15]: Please, add spaces in favour of code readability  !!!

File: test/pushito_test.exs [hunk 3]
```diff
@@ -2,7 +2,30 @@ defmodule PushitoTest do
...
+    apple_host = Application.fetch_env!(:pushito, :apple_host)
+    connection_name = :connection1
+    config = %Pushito.Config{:name => connection_name,
+                             :type => :cert,
+                             :apple_host => apple_host,
+                             :cert_file => cert_file,
+                             :key_file => key_file}
+
+    {:ok, pid} = Pushito.connect(config)
+    ^pid = Process.whereis connection_name
```
Comment (cabol) [inline test/pushito_test.exs]: Why not to use asserts  macros?? For example: `assert Process.whereis connection_name == pid`

File: test/pushito_test.exs [hunk 4]
```diff
@@ -2,7 +2,30 @@ defmodule PushitoTest do
...
+    :ok = Pushito.close connection_name
+    false = Process.alive? pid
```
Comment (cabol) [inline test/pushito_test.exs]: Same comment above. Example: `refute Process.alive?`

File: README.md [hunk 5]
```diff
@@ -1,2 +1,22 @@
 # pushito
-APNS over HTTP/2
+APNS over HTTP/2 for Elixir. Inspired on Erlang the project [apns4erl](https://github.com/inaka/apns4erl)
+
+**note** this project is under development
+
+## Tests
```
Comment (cabol) [inline README.md:6]: You should consider to create an integration test folder to do this – Check [Ecto](https://github.com/elixir-ecto/ecto)

**Review by amilkr — APPROVED** [review]

**Pull request merged**